WSLDiskShrinker/Scanner.cs

```diff
@@ -19,10 +19,11 @@ private static IEnumerable<ExactDistro> GetDistrosByRegistry()
 		var distroKeys = from distroID in distroIDs
 						 select lxss.OpenSubKey(distroID);
 		return from distroKey in distroKeys
-			   where Directory.Exists((string)distroKey.GetValue("BasePath"))
+			   let path = Path.Combine(distroKey.GetValue("BasePath") as string ?? string.Empty, "ext4.vhdx")
+			   where File.Exists(path)
```

Comment:
```suggestion
			   let basePath = distroKey.GetValue("BasePath") as string
			   where !string.IsNullOrWhiteSpace(basePath)
			   let path = Path.Combine(basePath, "ext4.vhdx")
			   where File.Exists(path)
```

We need to check if `BasePath` exists/is empty first before `Path.Combine`. Otherwise we might end up checking whether we have a `ext4.vhdx` file in the current directory.

Reply:
A good suggestion.